### PR TITLE
Feature/e UI 3842 cookie banner third parties aa

### PR DESCRIPTION
**Pull request by adnanakgun**
Resolves # https://tools.hmcts.net/jira/browse/EUI-3842

### CODE REVIEW

File: projects/exui-common-lib/src/lib/services/cookie/cookie.service.ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    this.document.cookie
+      .split('; ')
+      .filter(row => row.startsWith(`${key}`))
+      .forEach(element => {
+        this.deleteCookie(element.split('=')[0], path, domain);
+      });
+  }
+
+  public checkCookie(key: string): boolean {
+    return this.document.cookie.split(';').some(item => item.trim().startsWith(`${key}=`));
```
Comment (rayliangatos) [inline projects/exui-common-lib/src/lib/services/cookie/cookie.service.ts:42]: no space after ; here, which one is correct?

File: projects/exui-common-lib/src/lib/components/cookie-banner/cookie-banner.component.html [hunk 2]
```diff
@@ -0,0 +1,25 @@
+<div class="govuk-cookie-banner " role="region" attr.aria-label="'Cookies on {{appName}}'" *ngIf="isCookieBannerVisible">
+  <div class="govuk-cookie-banner__message govuk-width-container">
+
+    <div class="govuk-grid-row">
+      <div class="govuk-grid-column-two-thirds">
+        <h2 class="govuk-cookie-banner__heading govuk-heading-m">Cookies on {{appName}}</h2>
+
+        <div class="govuk-cookie-banner__content">
+          <p>We use some essential cookies to make this service work.</p>
+          <p>We’d also like to use analytics cookies so we can understand how you use the service and make improvements.</p>
```
Comment (phillip-whitaker) [inline projects/exui-common-lib/src/lib/components/cookie-banner/cookie-banner.component.html:10]: Did this content come from somewhere - it doesn't read very well.

Comment (adnanakgun) [reply projects/exui-common-lib/src/lib/components/cookie-banner/cookie-banner.component.html:10]: GDS

File: projects/exui-common-lib/src/lib/services/cookie/cookie.service.ts [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+  public deleteCookie(key: string, path?: string, domain?: string): void {
+    const pathValue = path ? `; path=${path}` : '';
+    const domainValue = domain ? `; domain=${domain}` : '';
+    this.document.cookie = `${key}=${pathValue}${domainValue}; expires=Thu, 01 Jan 1970 00:00:01 GMT; max-age=0`;
+  }
+
+  public deleteCookieByPartialMatch(key: string, path?: string, domain?: string): void {
+    this.document.cookie
+      .split('; ')
```
[the remark this answers is not in the document]
Comment (adnanakgun) [reply projects/exui-common-lib/src/lib/services/cookie/cookie.service.ts:34]: I don't think it makes much difference but I space makes more sense to me, document.cookie is essentially a string.

File: projects/exui-common-lib/src/lib/services/cookie/cookie.service.ts (hunk 1 above)
Comment (adnanakgun) [reply projects/exui-common-lib/src/lib/services/cookie/cookie.service.ts]: I've added a space here

File: projects/exui-common-lib/src/lib/components/cookie-banner/cookie-banner.component.ts [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+  }
+
+  public notifyAcceptance(): void {
+    this.acceptanceNotifier.emit();
+  }
+
+  private getExpiryDate(): string {
+    const now = new Date();
+    const time = now.getTime();
+    const expireTime = time + 3600 * 1000 * 24 * 365; // in 365 days
```
[the remark this answers is not in the document]
Comment (adnanakgun) [reply projects/exui-common-lib/src/lib/components/cookie-banner/cookie-banner.component.ts]: Made the change

**Review by rayliangatos — APPROVED** [review]

**Review by phillip-whitaker — APPROVED** [review]
LGBTQ+

**Pull request merged**